NoomBot/br/api/wapi.php

```diff
@@ -367,7 +367,11 @@ function getpagelimit() {
 	}
 
 	function getToken($page = null, $type = 'edit') {
-		$actions = array('action' => 'tokens', 'type' => $type);
+		if ($type == 'edit') {
+			$type = 'csrf';
+		}
+
+		$actions = array('action' => 'query', 'meta' => 'tokens', 'type' => $type);
 		$x = $this->query($actions, true);
 		$t = $x['tokens'][$type.'token'];
```

Comment:
This won't be right now

https://commons.wikimedia.org/w/api.php?action=tokens&type=edit

```
{
    "warnings": {
        "main": {
            "*": "Subscribe to the mediawiki-api-announce mailing list at <https://lists.wikimedia.org/postorius/lists/mediawiki-api-announce.lists.wikimedia.org/> for notice of API deprecations and breaking changes. Use [[Special:ApiFeatureUsage]] to see usage of deprecated features by your application."
        },
        "tokens": {
            "*": "\"action=tokens\" has been deprecated. Please use \"action=query&meta=tokens\" instead."
        }
    },
    "tokens": {
        "edittoken": "REDACTED+\\"
    }
}
```

vs https://commons.wikimedia.org/w/api.php?action=query&meta=tokens&type=csrf

```
{
    "batchcomplete": "",
    "query": {
        "tokens": {
            "csrftoken": "REDACTED+\\"
        }
    }
}
```

Presumably needs to be `$t = $x['query']['tokens'][$type.'token'];` to take account of the query part now in the result

Reply:
I agree that this is wrong, but I can't get github to let me add another commit to this, not sure why

Reply:
Because you've deleted the repo/fork.

Reply:
> Because you've deleted the repo/fork.

Oh, oops. Is there any way to restore it? Or should I abandon this merge request and create a new one?

Reply:
https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/restoring-a-deleted-repository might work. I haven't tried though.

Reply:
No recoverable repositories were found for DannyS712.

Is it possible for you to use the "suggest changes" for a specific change that I can then approve and incorporate? I've seen some documentation for that but cannot suggest my own such change (I assume because its my patch)

Reply:
```suggestion
		$t = $x['query']['tokens'][$type.'token'];
```

Reply:
That can probably only be used by someone approving the code though...

Reply:
Hmm, yeah, it doesn't give me an option. I have to go, but I can submit a new patch that fixes this when I get back (and I'll remember not to delete the fork)